src/promptflow-core/promptflow/executor/_tool_resolver.py

```diff
@@ -462,9 +462,21 @@ def _resolve_llm_connection_with_provider(connection):
             # This is a fallback for the case that ServerlessConnection related tool is not ready
             # in legacy versions, then we can directly use OpenAIConnection.
             if isinstance(connection, ServerlessConnection):
+                # The ServerlessConnection should be passed into promptflow-tools as it is.
```

Comment:
Also add this comment:  This is a short-term solution. After MaaS supports OpenAI compatible rest API, we should remove the V1 tricky here.

Reply:
added